packages/tests/tests/connection_manager.spec.ts

```diff
@@ -0,0 +1,196 @@
+        this.timeout(TEST_TIMEOUT);
+
+        // simulate that the peer is not connected
+        getConnectionsStub.returns([]);
+
+        // simulate that the peer is a bootstrap peer
+        getTagNamesForPeerStub.resolves([Tags.BOOTSTRAP]);
+
+        // emit a peer:discovery event
+        waku.libp2p.dispatchEvent(
+          new CustomEvent("peer:discovery", { detail: "bootstrap-peer" })
+        );
+
+        // wait for the async function calls within attemptDial to finish
+        await delay(DELAY_MS);
+
+        // check that dialPeer was called once
+        expect(dialPeerStub.callCount).to.equal(
+          1,
+          "dialPeer should be called for bootstrap peers"
+        );
+      });
+
+      it("should not be called more than DEFAULT_MAX_BOOTSTRAP_PEERS_ALLOWED times for bootstrap peers", async function () {
+        this.timeout(TEST_TIMEOUT);
+
+        // simulate that the peer is not connected
+        getConnectionsStub.returns([]);
+
+        // simulate that the peer is a bootstrap peer
+        getTagNamesForPeerStub.resolves([Tags.BOOTSTRAP]);
+
+        // emit first peer:discovery event
+        waku.libp2p.dispatchEvent(
+          new CustomEvent("peer:discovery", { detail: "bootstrap-peer" })
+        );
+
+        // simulate that the peer is connected
+        getConnectionsStub.returns([{ tags: [{ name: Tags.BOOTSTRAP }] }]);
+
+        // emit multiple peer:discovery events
+        const totalBootstrapPeers = 5;
+        for (let i = 1; i <= totalBootstrapPeers; i++) {
+          await delay(500);
+          waku.libp2p.dispatchEvent(
+            new CustomEvent("peer:discovery", {
+              detail: `bootstrap-peer-id-${i}`,
+            })
+          );
+        }
+
+        // check that dialPeer was called only once
+        expect(dialPeerStub.callCount).to.equal(
+          1,
+          "dialPeer should not be called more than once for bootstrap peers"
+        );
+      });
+    });
+
+    describe("For peer-exchange peers", function () {
+      it("should be called for peers with PEER_EXCHANGE tags", async function () {
+        this.timeout(TEST_TIMEOUT);
+
+        // simulate that the peer is not connected
+        getConnectionsStub.returns([]);
+
+        // simulate that the peer has a PEER_EXCHANGE tag
+        getTagNamesForPeerStub.resolves([Tags.PEER_EXCHANGE]);
+
+        // emit a peer:discovery event
+        waku.libp2p.dispatchEvent(
+          new CustomEvent("peer:discovery", { detail: "px-peer" })
+        );
+
+        // wait for the async function calls within attemptDial to finish
+        await delay(DELAY_MS);
+
+        // check that dialPeer was called once
+        expect(dialPeerStub.callCount).to.equal(
+          1,
+          "dialPeer should be called for peers with PEER_EXCHANGE tags"
+        );
+      });
+
+      it("should be called for every peer with PEER_EXCHANGE tags", async function () {
```

Comment:
Are we checkng that peer exhcange peers are not limited by the max bootstrap peers?